src/components/autocomplete.tsx

```diff
@@ -0,0 +1,181 @@
+import { useCallback, useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import AutocompleteItem, { AutocompleteItemType } from './autocomplete-item';
+import SearchInput from './search-input';
+
+import { getLastIndexOfSubstringIgnoreCase } from '../utils';
+
+import '../styles/components/dropdown.scss';
+import '../styles/components/autocomplete.scss';
+
+export const filterOptions = (items: AutocompleteItemType[], query: string) =>
+  items.filter(
+    (item) => getLastIndexOfSubstringIgnoreCase(item.pathLabel, query) >= 0
+  );
+
+export const shouldShowDropdown = (
+  textInputValue: string,
+  data: AutocompleteItemType[],
+  selected: boolean,
+  filter: boolean,
+  minCharsToShowDropdown: number
+) => {
+  const trimmed = textInputValue.trim();
+  let showDropdown = false;
+  if (trimmed && !selected && trimmed.length >= minCharsToShowDropdown) {
+    const found = filter ? filterOptions(data, trimmed) : data;
+    showDropdown = found.length > 0;
+  }
+  return showDropdown;
+};
+
+type AutocompleteProps = {
+  data: AutocompleteItemType[];
+  onSelect: (selected: AutocompleteItemType | string) => void;
+  onChange?: (textInput: string) => void;
+  showDropdownUpdated?: (updated: boolean) => void;
+  clearOnSelect?: boolean;
+  placeholder?: string;
+  filter?: boolean;
+  value?: string;
+  minCharsToShowDropdown?: number;
+  isLoading?: boolean;
+  autoFocus?: boolean;
+};
+
+const Autocomplete = ({
+  data,
+  onSelect,
+  onChange,
+  showDropdownUpdated,
+  clearOnSelect = false,
+  placeholder = '',
+  filter = true,
+  value = '',
+  minCharsToShowDropdown = 0,
+  isLoading = false,
+  autoFocus = false,
+}: AutocompleteProps) => {
```

Comment:
Would probably be good to have `...props` and pass it down either to the underlying `div`, or maybe better to the `SearchInput` component

Reply:
done